modules/social_features/social_follow_taxonomy/modules/social_follow_tag/social_follow_tag.module

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/**
+ * @file
+ * Contains social_follow_tag.module.
+ */
+
+use Drupal\Core\Url;
+use Drupal\flag\Entity\Flag;
+use Drupal\flag\FlagInterface;
+use Drupal\taxonomy\TermInterface;
+
+/**
+ * Implements hook_theme().
+ */
+function social_follow_tag_theme() {
+  return [
+    'paragraph__tag__default' => [
+      'base hook' => 'paragraph',
+    ],
+    'social_tagging_nosplit' => [
+      'base hook' => 'social_tagging_nosplit',
+    ],
+    'social_tagging_split' => [
+      'base hook' => 'social_tagging_split',
+    ],
+    'search_follow_tag' => [
+      'variables' => ['tags' => NULL],
+    ],
+  ];
+
+}
+
+/**
+ * Prepares variables for the Tag paragraph template.
+ *
+ * Implements hook_preprocess_HOOK().
+ * {@inheritdoc}
+ */
+function social_follow_tag_preprocess_paragraph(&$variables) {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $entity */
+  $entity = $variables['elements']['#paragraph'];
+  $bundle = $entity->bundle();
+
+  switch ($bundle) {
+    case 'tag':
+      if ($entity->hasField('field_tag') && !empty($entity->field_tag)) {
+        $term_id = $entity->field_tag->target_id;
+        /** @var \Drupal\taxonomy\TermStorageInterface $term_storage */
+        $term_storage = \Drupal::entityTypeManager()->getStorage('taxonomy_term');
+        /** @var \Drupal\taxonomy\TermInterface $term */
+        $term = $term_storage->load($term_id);
+
+        if (!empty($term)) {
+          // Creates a link to a search page with a tag as a filter parameter.
+          $parameter = 'tag';
+          $route = 'view.search_content.page_no_value';
+          $parents = $term_storage->loadParents($term_id);
+          if (!empty($parents)) {
+            $parent = reset($parents);
+            $category = $parent->getName();
+
+            if ($tag_service = Drupal::getContainer()->get('social_tagging.tag_service')->allowSplit()) {
+              $parameter = social_tagging_to_machine_name($category);
+            }
+          }
+          $url = Url::fromRoute($route, [
+            $parameter . '[]' => $term_id,
+          ]);
+
+          // Adding additional data to the term variable to extend the template.
+          $variables['term'] = [
+            'url' => $url->toString(),
+            'name' => $term->getName(),
+            'flag' => _social_follow_taxonomy_flag_link($term),
+            'related_content' => _social_follow_tag_related_content($term),
+          ];
+        }
+      }
+      break;
+  }
+}
+
+/**
+ * Extends variables for social_tagging_split template.
+ *
+ * Implements hook_preprocess_HOOK().
+ * {@inheritdoc}
+ */
+function social_follow_tag_preprocess_social_tagging_split(&$variables) {
+  // Iterate over an array with parent terms (category).
+  foreach ($variables['taghierarchy'] as $parent_tag) {
```

Comment:
@VolodymyrMartsynenko  Are we sure that an array always will be passed to `$variables['taghierarchy']`?
If not - maybe good to add some additional check here.

Reply:
I added an additional check for this.